packages/web/src/pages/bulletins/bulletinList.tsx

```diff
@@ -19,17 +22,32 @@ interface FilteredInfoState {
 }
 
 const BulletinList = () => {
+  const query = useQuery();
   const [filteredInfo, setFilteredInfo] = React.useState<FilteredInfoState>({});
   let timeout: any;
 
+  const { push } = useHistory();
+
+  const searchQuery = query.get("search") || "";
+  const searchInputFieldRef = useRef<Input>(null);
+
   const handleRaceFilter = (filters: any) => {
     setFilteredInfo({ ...filteredInfo, races: filters.races });
   };
 
+  const changeRoute = (searchValue: string) => {
+    push({
+      pathname: routes.bulletinsBase(),
+      search: `?search=${searchValue}`,
+    });
+  };
+
   const handleNameSearch = (e: ChangeEvent<HTMLInputElement>) => {
     clearTimeout(timeout);
     timeout = setTimeout(() => {
-      handleSearch(e.target.value, "bulletinName");
+      const targetValue = e.target.value;
+      changeRoute(targetValue);
+      handleSearch(targetValue, "bulletinName");
```

Comment:
Just an FYI to keep an eye one thing and that's that handle search is triggered twice on change. 
It's triggered here and than when the route is changed again in the useEffect on line 64

However I think it's not a problem in this case and I think it's better. Because when we would do only change the route and we would not trigger handle search here it would work. It would go -> change the url -> useEffect -> handle search. And the "reaction of the components" would be slower. 

Btw this is just FYI to keep an eye on this - in some other cases it could cause an problem but I think here it's completely OK.

Reply:
Noted, thanks!